File: cpp/Classes/bots/PushTheTempoBot.cpp
#include "PushTheTempoBot.h"

sample_game::PushTheTempoBot::PushTheTempoBot() {
    _randomSource.seed(std::random_device{}());
}

sample_game::Move sample_game::PushTheTempoBot::calcNextMove(const std::vector<std::vector<int>>& gameBoardState,
                                                               const std::unordered_set<Move>& availableMoves) {
    if (availableMoves.empty())
        return Move::INCORRECT_MOVE;

    // make random available move
    std::vector<Move> availableMovesToSelect(availableMoves.begin(), availableMoves.end());
    std::shuffle(std::begin(availableMovesToSelect), std::end(availableMovesToSelect), _randomSource);
    return availableMovesToSelect[0];
}

File: csharp_unity/Assets/Src/Bots/PushTheTempoBot.cs
using System.Collections.Generic;
using System.Linq;
using sample_game.utils;

namespace sample_game {
    
    /// <summary>
    /// Simple bot that makes random available moves.
    /// </summary>
    public class PushTheTempoBot : IBot {
        
        //-------------------------------------------------------------
        // Nested
        //-------------------------------------------------------------

        //-------------------------------------------------------------
        // Dependencies
        //-------------------------------------------------------------

        //-------------------------------------------------------------
        // Class constants
        //-------------------------------------------------------------

        //-------------------------------------------------------------
        // Class variables
        //-------------------------------------------------------------
        
        //-------------------------------------------------------------
        // Class methods
        //-------------------------------------------------------------

        //-------------------------------------------------------------
        // Constructor/destructor
        //-------------------------------------------------------------
        
        //-------------------------------------------------------------
        // Variables
        //-------------------------------------------------------------

        //-------------------------------------------------------------
        // Events
        //-------------------------------------------------------------

        //-------------------------------------------------------------
        // Properties
        //-------------------------------------------------------------

        //-------------------------------------------------------------
        // Private properties serialized for Unity
        //-------------------------------------------------------------

        //-------------------------------------------------------------
        // Public methods
        //-------------------------------------------------------------

        public Move CalcNextMove(List<List<int>> gameBoardState, HashSet<Move> availableMoves) {
            return availableMoves.Count > 0
                ? availableMoves.Shuffle().First()
                : Move.IncorrectMove;
        }
        
        //-------------------------------------------------------------
        // Protected methods
        //-------------------------------------------------------------

        //-------------------------------------------------------------
        // Private methods
        //-------------------------------------------------------------

        //-------------------------------------------------------------
        // Unity methods
        //-------------------------------------------------------------

        //-------------------------------------------------------------
        // Handlers
        //-------------------------------------------------------------
    }
} // namespace sample_game